packages/xml/src/decorators.ts

```diff
@@ -19,11 +20,25 @@ import type { XmlNamespace } from "./types.js";
 /** @internal */
 export const namespace = "TypeSpec.Xml";
 
+const [_getXmlName, setXmlName] = useStateMap<Type, string>(XmlStateKeys.xmlName);
+
 /** {@inheritDoc NameDecorator}  */
 export const $name: NameDecorator = (context, target, name) => {
+  setXmlName(context.program, target, name);
   context.call($encodedName, target, "application/xml", name);
 };
 
+/**
+ * Get the XML name for the given type. Returns undefined if no XML name is set.
+ *
+ * @param program The TypeSpec program.
+ * @param target The target type.
+ * @returns The XML name, or undefined if not set.
+ */
+export function getXmlName(program: Program, target: Type): string | undefined {
```

Comment:
the problem with this is this won't cover if the xml name was set via `@encodedName`

The way we check if you need xml.name in openapi3 emitter is if the name is different from the original one